src/main/java/uk/ac/ebi/embl/api/validation/dao/EraproDAOUtilsImpl.java

```diff
@@ -114,113 +115,91 @@ public EraproDAOUtilsImpl (Connection connection)
 	{
 		this.connection=connection;
 	}
-	
+
+
+	private String getAddress(String analysisId) throws ValidationEngineException {
+
+		String addressQuery = "select broker_name, a.center_name, sa.laboratory_name, sa.address, sa.country "
+				+ "from analysis a "
+				+ "join submission_account sa using(submission_account_id) "
+				+ "where a.analysis_id =?";
+
+		PreparedStatement addressStmt = null;
+		ResultSet addressRs = null;
+
+		try {
+			addressStmt = connection.prepareStatement(addressQuery);
+			addressStmt.setString(1, analysisId);
+			addressRs = addressStmt.executeQuery();
+			if (addressRs.next()) {
+				ReferenceEntity refEntity = new ReferenceEntity();
+				refEntity.setBrokerName(addressRs.getString("broker_name"));
+				refEntity.setCenterName(addressRs.getString("center_name"));
+				refEntity.setLaboratoryName(addressRs.getString("laboratory_name"));
+				refEntity.setAddress(addressRs.getString("address"));
+				refEntity.setCountry(addressRs.getString("country"));
+				return new ReferenceUtils().getAddress(refEntity);
+			}
+		} catch(SQLException | UnsupportedEncodingException e ) {
+			throw new ValidationEngineException(e);
+		} finally {
+			DbUtils.closeQuietly(addressRs);
+			DbUtils.closeQuietly(addressStmt);
+		}
+
+		return null;
+	}
+
 	@Override
-	public Reference getSubmitterReference(String analysisId) throws SQLException, UnsupportedEncodingException
-    {
-		if(submitterReferenceCache.get(analysisId)!=null)
-    	{
-    		return submitterReferenceCache.get(analysisId);
-    	}
-    	Publication publication = new Publication();
-		ReferenceFactory referenceFactory = new ReferenceFactory();
-		Reference reference = referenceFactory.createReference();
-		HashSet<String> consortium=new HashSet<String>();
-		String pubConsortium="";
-		String submitterReferenceQuery = "select sc.consortium, sc.surname, sc.middle_initials, sc.first_name, s.center_name, to_char(s.first_created, 'DD-MON-YYYY') first_created, sa.laboratory_name, sa.address, sa.country "
-				              + "from analysis a "
-				              + "join submission s on(s.submission_id=a.submission_id) "
-				              + "join submission_account sa on(s.submission_account_id=sa.submission_account_id) "
-				              + "join submission_contact sc on(sc.submission_account_id=s.submission_account_id) "
-				              + "where a.analysis_id =?";
-		
+	public Reference getSubmitterReference(String analysisId) throws SQLException, UnsupportedEncodingException {
+		return new ReferenceUtils().constructReference(fetchSubmitterReference(analysisId));
+	}
+
+
+	private List<ReferenceEntity> fetchSubmitterReference(String analysisId) throws SQLException, UnsupportedEncodingException
+	{
+		List<ReferenceEntity> referenceEntityList = new ArrayList<>();
+
+		String submitterReferenceQuery = " select broker_name,a.submission_account_id, sc.consortium, sc.surname, sc.middle_initials, " +
```

Comment:
- We have copy of submission account info for each submission contact
- WOuld be easier to have two selects one for submission contact and another for submission account and have submitterReference class with ONE submission account  and ONE OR MORE submission contact